### ISSUE TITLE
8.2.0 rc Umbraco Forms Macro Insert Form into Rich Text Editor Bug

### ISSUE THREAD

**Issue by prjseal** [umbraco/Umbraco-CMS#6576]
Using the 8.2.0 rc zip with VS Code and IIS Express

I installed the clean starter kit

I installed Umbraco Forms

I created  a sample form

I tried to add the form in a rich text editor

![https://i.imgur.com/qjIqjqk.jpg](https://i.imgur.com/qjIqjqk.jpg)

I got the following error:

`Failed to retrieve macro result for macro with alias renderUmbracoForm`

![https://i.imgur.com/XyR320P.jpg](https://i.imgur.com/XyR320P.jpg)

**Comment by nul800sebastiaan** [comment]
Hmm, interesting! We'll have to have a look. Do you happen to have the error that you can see in the network request?

**Comment by prjseal** [comment]
```
Possibly unhandled rejection: {"errorMsg":"Failed to retrieve macro result for macro with alias  renderUmbracoForm","data":{"Message":"An error has occurred.","ExceptionMessage":"Error executing child request for handler 'System.Web.Mvc.HttpHandlerUtil+ServerExecuteHttpHandlerAsyncWrapper'.","ExceptionType":"System.Web.HttpException","StackTrace":"   at System.Web.HttpServerUtility.ExecuteInternal(IHttpHandler handler, TextWriter writer, Boolean preserveForm, Boolean setPreviousPage, VirtualPath path, VirtualPath filePath, String physPath, Exception error, String queryStringOverride)\r\n   at System.Web.HttpServerUtility.Execute(IHttpHandler handler, TextWriter writer, Boolean preserveForm, Boolean setPreviousPage)\r\n   at System.Web.HttpServerUtility.Execute(IHttpHandler handler, TextWriter writer, Boolean preserveForm)\r\n   at System.Web.Mvc.Html.ChildActionExtensions.ActionHelper(HtmlHelper htmlHelper, String actionName, String controllerName, RouteValueDictionary routeValues, TextWriter textWriter)\r\n   at System.Web.Mvc.Html.ChildActionExtensions.RenderAction(HtmlHelper htmlHelper, String actionName, String controllerName, Object routeValues)\r\n   at ASP._Page_Views_MacroPartials_InsertUmbracoFormWithTheme_cshtml.Execute() in d:\\piess\\Code\\Umbraco Versions\\UmbracoCms.8.2.0-rc\\Views\\MacroPartials\\InsertUmbracoFormWithTheme.cshtml:line 49\r\n   at System.Web.WebPages.WebPageBase.ExecutePageHierarchy()\r\n   at System.Web.Mvc.WebViewPage.ExecutePageHierarchy()\r\n   at System.Web.WebPages.WebPageBase.ExecutePageHierarchy(WebPageContext pageContext, TextWriter writer, WebPageRenderingBase startPage)\r\n   at Umbraco.Web.Mvc.ProfilingView.Render(ViewContext viewContext, TextWriter writer)\r\n   at Umbraco.Web.Mvc.ControllerExtensions.RenderViewResultAsString(ControllerBase controller, ViewResultBase viewResult)\r\n   at Umbraco.Web.Macros.PartialViewMacroEngine.Execute(MacroModel macro, IPublishedContent content)\r\n   at Umbraco.Web.Macros.MacroRenderer.<>c__DisplayClass17_0.<ExecuteMacroOfType>b__0()\r\n   at Umbraco.Web.Macros.MacroRenderer.ExecuteProfileMacroWithErrorWrapper(MacroModel macro, String msgIn, Func`1 getMacroContent, Func`1 msgErr)\r\n   at Umbraco.Web.Macros.MacroRenderer.ExecuteMacroWithErrorWrapper(MacroModel macro, String msgIn, String msgOut, Func`1 getMacroContent, Func`1 msgErr)\r\n   at Umbraco.Web.Macros.MacroRenderer.ExecuteMacroOfType(MacroModel model, IPublishedContent content)\r\n   at Umbraco.Web.Macros.MacroRenderer.Render(MacroModel macro, IPublishedContent content, IDictionary pageElements)\r\n   at Umbraco.Web.UmbracoComponentRenderer.RenderMacro(String alias, IDictionary`2 parameters, IPublishedContent content)\r\n   at Umbraco.Web.Editors.MacroRenderingController.GetMacroResultAsHtml(String macroAlias, Int32 pageId, IDictionary`2 macroParams)\r\n   at lambda_method(Closure , Object , Object[] )\r\n   at System.Web.Http.Controllers.ReflectedHttpActionDescriptor.ActionExecutor.<>c__DisplayClass6_2.<GetExecutor>b__2(Object instance, Object[] methodParameters)\r\n   at System.Web.Http.Controllers.ReflectedHttpActionDescriptor.ExecuteAsync(HttpControllerContext controllerContext, IDictionary`2 arguments, CancellationToken cancellationToken)\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Controllers.ApiControllerActionInvoker.<InvokeActionAsyncCore>d__1.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Filters.ActionFilterAttribute.<ExecuteActionFilterAsyncCore>d__5.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Filters.ActionFilterAttribute.<CallOnActionExecutedAsync>d__6.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n 
```

**Comment by bjornnorlen** [comment]
We also experience this issue using Umbraco 8.2.1 and Umbraco forms installed from the backoffice with version 8.2.0!

**Comment by andersbrohall** [comment]
+1

**Comment by Rickard-Lovgren** [comment]
+1 , Forms 8.2.0 and Umbraco 8.4.0

Same error when inserting a Form in the Grid, it still works though and it get's rendered in view mode. In admin it still states "click to choose a form..." but when clicking the one I previously chosed is there. It seems that it's only the rendering in edit mode that doesn't work.

**Comment by sstottelaar** [comment]
I also experience this issue using UmbracoForms 8.2.0 and Umbraco 8.4.0. I can't add a form using the grid layout editor.

**Comment by tombmedia** [comment]
+1 same error inserting a form into the RTE forms 8.2 and umbraco 8.4

**Comment by nul800sebastiaan** [comment]
Thanks everyone for the feedback! Sorry for the late reply! 

I just learned that this problem was fixed already, and I completely missed it - the issue + hotfix can be found here: https://github.com/umbraco/Umbraco.Forms.Issues/issues/235#issuecomment-537412133

The fix will be released in Forms 8.2.1, not sure what the release date is for that yet. 

Hope this helps.
